hop/io.py

```diff
@@ -111,11 +114,32 @@ def open(self, url, mode="r", metadata=False):
             raise ValueError("mode must be either 'w' or 'r'")
 
 
-class Deserializer(Enum):
-    CIRCULAR = models.GCNCircular
-    VOEVENT = models.VOEvent
-    BLOB = models.MessageBlob
+def _load_deserializer_plugins():
+    """Load all registered deserializer plugins.
+
+    """
+    # set up plugin manager
+    manager = pluggy.PluginManager("hop")
+    manager.add_hookspecs(plugins)
+
+    # load in base models
+    manager.register(models)
+
+    # load external models
+    try:
+        manager.load_setuptools_entrypoints("hop_plugin")
+    except Exception:
+        logger.warning("WARNING: could not load external hop models, verify installed plugins")
 
+    # add all registered plugins to registry
+    registered = {}
+    for model_plugins in manager.hook.get_models():
+        registered.update({name.upper(): model for name, model in model_plugins.items()})
+
```

Comment:
Personally, I find this a bit more clear (but just a bit):
```python
registered = {}
for model_plugins in manager.hook.get_models():
    for name, model in model_plugins.items():
        registered[name.upper()] = model
```

---

I think we should warn when there are name collisions, not silently pick the last one. That'd be a nasty thing to try to debug.

Reply:
Will do, I'll update that code snippet with your suggestion. Regarding the name collision, that's a good idea. I'll add a check to see if the key exists while looping over the various plugins and warn if there are duplicates.